### ISSUE TITLE
How does inbound-mode client know if drachtio "gives up" on a transaction

### ISSUE THREAD

**Issue by elbow** [davehorton/drachtio-server#132]
Hi Dave,

I need to implement the tracking of active calls in my drachtio-srf using client.

I see that drachtio "gives up" on a transaction after a minute with nothing happening:

```
2020-06-07 12:16:02.285665 pending-request: running timer function
2020-06-07 12:16:02.285776 pending-request: timer not set (queue is empty after processing expired timers), length: 0
2020-06-07 12:16:02.285812 PendingRequestController::timeout: giving up on transactionId 1dba7196-6138-41aa-923f-6f463b63621a
2020-06-07 12:16:02.285869 ClientController::removeNetTransaction: transactionId 1dba7196-6138-41aa-923f-6f463b63621a; size: 0
```

Is there an event or some other way to detect this in my client? 

Since after this the cancel from the caller side isn't delivered to my client and so I have a call that I think is still active in the client but actually it is gone.

Thanks.

**Comment by davehorton** [comment]
Well this timeout indicates an error of some kind....it means it sent an invite over to a drachtio client and never got a response back as to how to handle the invite.  That should not ever be happening in a normal scenario.  Is there some scenario where your client app does not respond/handle the incoming invite for some reason?

**Comment by elbow** [comment]
Hi,

Yes - the drachtio client was trying to wake up the client app using push, and the push failed to work.

So we had no progress messages to send back to Drachtio.

Meanwhile the caller side just kept waiting past the 1 minute timeout here.

Would it help if I made my Drachtio client send a 100 Trying back as it starts?

I did fix my code such that it sends a 500 back if the push fails.  So that will resolve this problem.

Thanks

**Comment by davehorton** [comment]
No, the drachtio server will have already sent the 100 Trying -- that must go out immediately.

You should probably send something like a 180 Ringing while you are trying to wake the client

**Comment by elbow** [comment]
Hi Dave - I'd prefer not to send a 180 unless the end device is actually ringing.  But let me give it proper thought since the push process can be quite slow.

Thanks for the replies.
